### PR TITLE
CORDA-1725 Add documentation on building Corda for Win/*nix and clean up doc bui…

### PR DESCRIPTION
**Pull request by anthonykeenan**
Complete list of everything needed on a new machine to build Corda on Windows/Ubuntu Linux. Someone with a Mac will have to add instructions for Mac seperately.

### CODE REVIEW

**Comment by anthonykeenan** [comment]
I've added you as my guinea-pig Windows user @vkolomeyko

**Comment by vkolomeyko** [comment]
Looks fine to me... however I would separate installation requirements for building Corda and building Corda Docs.
If someone decides to build Corda distribution, but not the Docs, my understanding that all this Python toolkit will not be necessary.

**Comment by anthonykeenan** [comment]
I believe it does run the build docs as part of a `gradle build`, hence why I put them together, but I will double check.

**Comment by vkolomeyko** [comment]
`gradle build` also runs unit tests... are we sure we needs this when we are aiming just to build Corda?

In the past to perform the build I was running `gradlew.bat clean install`.

**Comment by anthonykeenan** [comment]
`gradle build` should be the target we encourage everyone to use as part of a PR - we _want_ them to run the unit tests surely? But I think you're right, `gradle build` doesn't run the docs... although I'm sure it was doing. Maybe someone changed it. I'll do a bit more experimentation and split them back out.

**Comment by vkolomeyko** [comment]
I guess which Gradle task to run is also a function of why people do the build.
Possible options I see:
1. Create their own binaries to later on develop upon or run Demos;
2. Make a code contribution to Corda with a view to have it incorporated into future Corda releases.

File: docs/source/building-corda.rst [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+5. Download and run the executable to install Java (use the default settings)
+6. Add Java to the PATH environment variable by following the instructions at https://docs.oracle.com/javase/7/docs/webnotes/install/windows/jdk-installation-windows.html#path
+7. Open a new command prompt and run ``java -version`` to test that Java is installed correctly
+
+Git, bash and make
+~~~~~~~~~~~~~~~~~~
+
+In order to build Corda you need a ``bash`` emulator with ``make`` installed and accessible from the command prompt. Git for
+Windows ships with a version of MinGW that contains a ``bash`` emulator, to which you can download and add a Windows port of
+make, instructions for which are provided below. Alternatively you can install a full version of MinGW from `here <http://www.mingw.org/>`_.
```
Comment (tlil) [inline docs/source/building-corda.rst]:
```
``make``
```

File: docs/source/building-corda.rst [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+1. Visit https://git-scm.com/download/win
+2. Click the "64-bit Git for Windows Setup" download link.
+3. Download and run the executable to install Git (use the default installation values) and make a note of the installation directory.
+4. Open a new command prompt and type ``git --version`` to test that git is installed correctly
```
Comment (tlil) [inline docs/source/building-corda.rst]:
```
``git``
```

or Git

File: docs/source/building-corda.rst [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+Buillding Corda
+~~~~~~~~~~~~~~~
+
+1. Open a command prompt
+2. Run ``git clone https://github.com/corda/coda``
```
Comment (tlil) [inline docs/source/building-corda.rst]: `...corda/corda.git`

not coda :)

File: docs/source/building-corda.rst [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+2. Run ``sudo apt-get update``
+3. Then run ``sudo apt-get install oracle-java8-installer``. Press Y when prompted and agree to the licence terms.
+4. Run ``java --version`` to verify that java is installed correctly
+
+
+Building Corda
+~~~~~~~~~~~~~~
+
+1. Open the terminal
+2. Run ``git clone https://github.com/corda/coda``
```
Comment (tlil) [inline docs/source/building-corda.rst]: `.../corda/corda.git`

File: docs/source/building-the-docs.rst [hunk 5]
```diff
@@ -7,44 +7,109 @@ is pre-generated, as well as code documentation, and this can be done automatica
...
-.. sourcecode:: shell
+Git, bash and make
+~~~~~~~~~~~~~~~~~~
+
+In order to build the documentation for Corda you need a ``bash`` emulator with ``make`` installed and accessible from the command prompt. Git for
+Windows ships with a version of MinGW that contains a ``bash`` emulator, to which you can download and add a Windows port of
+make, instructions for which are provided below. Alternatively you can install a full version of MinGW from `here <http://www.mingw.org/>`_.
+
+1. Go to `ezwinports <https://sourceforge.net/projects/ezwinports/files/>`_ and click the download for ``make-4.2.1-without-guile-w32-bin.zip``
+2. Navigate to the git installation directory (by default ``C:\Program Files\Git``), open ``mingw64``
```
Comment (tlil) [inline docs/source/building-the-docs.rst]:
```
``git``
```

Comment (tlil) [reply docs/source/building-the-docs.rst]: ... or Git (there are a couple of other instances as well)

File: docs/source/building-the-docs.rst [hunk 6]
```diff
@@ -7,44 +7,109 @@ is pre-generated, as well as code documentation, and this can be done automatica
...
+   like ``make: *** No targets specified and no makefile found.  Stop.``)
+
+
+Python, pip and virtualenv
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+1. Visit https://www.python.org/downloads
+2. Scroll down to the most recent v2 release (tested with v.2.7.15) and click the download link
+3. Download the "Windows x86-64 MSI installer"
+4. Run the installation, making a note of the python installation directory (defaults to ``c:\Python27``)
```
Comment (tlil) [inline docs/source/building-the-docs.rst]: s/python/Python/g

File: docs/source/building-the-docs.rst [hunk 7]
```diff
@@ -7,44 +7,109 @@ is pre-generated, as well as code documentation, and this can be done automatica
...
+5. Add the python installation directory (e.g. ``c:\Python27``) to your system PATH environment variable
+6. Add the python scripts sub-directory (e.g. ``c:\Python27\scripts``) to your System PATH environment variable
```
Comment (tlil) [inline docs/source/building-the-docs.rst]: system

File: docs/source/building-the-docs.rst [hunk 8]
```diff
@@ -7,44 +7,109 @@ is pre-generated, as well as code documentation, and this can be done automatica
...
+1. Go to `ezwinports <https://sourceforge.net/projects/ezwinports/files/>`_ and click the download for ``make-4.2.1-without-guile-w32-bin.zip``
+2. Navigate to the git installation directory (by default ``C:\Program Files\Git``), open ``mingw64``
+3. Unzip the downloaded file into this directory, but do NOT overwrite/replace any existing files
+4. Add the git ``bin`` directory to your system PATH environment variable (by default ``C:\Program Files\Git\bin``)
+5. Open a new command prompt and run ``bash`` to test that you can access the git bash emulator
+6. Type ``make`` to make sure it has been installed successfully (you should get an error
+   like ``make: *** No targets specified and no makefile found.  Stop.``)
+
+
+Python, pip and virtualenv
```
Comment (tlil) [inline docs/source/building-the-docs.rst]: Pip, VirtualEnv (if you wanna be consistent with usage further down)

File: docs/source/building-the-docs.rst [hunk 9]
```diff
@@ -7,44 +7,109 @@ is pre-generated, as well as code documentation, and this can be done automatica
...
-documentation by running the following script:
+Once the requirements are installed, you can automatically build the HTML format user documentation, PDF, and
+the API documentation by running the following script:
 
 .. sourcecode:: shell
 
+    // On Windows
+    gradlew buildDocs
+
+    // On Mac
```
Comment (tlil) [inline docs/source/building-the-docs.rst]: or Linux

File: docs/source/building-the-docs.rst [hunk 10]
```diff
@@ -7,44 +7,109 @@ is pre-generated, as well as code documentation, and this can be done automatica
...
     ./gradlew buildDocs
 
-Alternatively you can build non-HTML formats from the ``docs`` folder. Change directory to the folder and then run the
-following to see a list of all available formats:
+Alternatively you can build non-HTML formats from the ``docs`` folder.
+
+However, running ``make`` from the command line requires further dependencies to be installed. When building in gradle they
```
Comment (tlil) [inline docs/source/building-the-docs.rst]: Gradle

**Pull request merged**